sql/src/test/java/io/crate/planner/operators/LogicalPlannerTest.java

```diff
@@ -0,0 +1,174 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.operators;
+
+import io.crate.analyze.QueryClause;
+import io.crate.analyze.SelectAnalyzedStatement;
+import io.crate.analyze.relations.QueriedRelation;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.analyze.symbol.format.SymbolPrinter;
+import io.crate.test.integration.CrateDummyClusterServiceUnitTest;
+import io.crate.testing.SQLExecutor;
+import org.hamcrest.FeatureMatcher;
+import org.hamcrest.Matcher;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.StringJoiner;
+
+import static org.hamcrest.Matchers.equalTo;
+
+public class LogicalPlannerTest extends CrateDummyClusterServiceUnitTest {
+
+    private SQLExecutor sqlExecutor;
+
+    @Before
+    public void prepare() {
+        sqlExecutor = SQLExecutor.builder(clusterService)
+            .enableDefaultTables()
+            .build();
+    }
+
+    private LogicalPlan plan(String statement) {
+        SelectAnalyzedStatement analyzedStatement = sqlExecutor.analyze(statement);
+        QueriedRelation relation = analyzedStatement.relation();
+        return LogicalPlanner.plan(relation)
+            .build(LogicalPlanner.extractColumns(relation.querySpec().outputs()))
+            .tryCollapse();
+    }
+
+    @Test
+    public void testSimpleSelectQAFAndLimit() throws Exception {
+        LogicalPlan plan = plan("select a from t1 order by a limit 10 offset 5");
+        assertThat(plan, isPlan("FetchOrEval[a] -> \n" +
+                                "Limit[10;5] -> \n" +
+                                "OrderBy[a] -> \n" +
+                                "Collect[doc.t1 | [a] | All]\n"));
+    }
+
+    @Test
+    public void testHavingGlobalAggregation() throws Exception {
+        LogicalPlan plan = plan("select min(a), min(x) from t1 having min(x) < 33 and max(x) > 100");
+        assertThat(plan, isPlan("FetchOrEval[min(a), min(x)] -> \n" +
+                                "Filter[((min(x) < 33) AND (max(x) > 100))] -> \n" +
+                                "Aggregate[min(a), min(x), max(x)] -> \n" +
+                                "Collect[doc.t1 | [a, x] | All]\n"));
+    }
+
+    @Test
+    public void testSelectCountStarIsOptimized() throws Exception {
+        LogicalPlan plan = plan("select count(*) from t1 where x > 10");
+        assertThat(plan, isPlan("FetchOrEval[count()] -> \n" +
+                                "Count[doc.t1 | (x > 10)]\n"));
```

Comment:
Not sure if we stick with the String matching. Would be nice to have Matchers for the Plan. However, this is really concise to look at.

Reply:
Maybe we can extend this later if the string representation becomes unwieldy. 
I wanted to avoid having to do the cast/assert properties dance like we've in the planner tests. Though maybe we could also come up with some kind of fluid matcher chaining api.

Like  `assertThat(plan, consistsOf(isFetchOrEvalOp(..), isCountOp(..)))`

Reply:
ftr: will keep it for now - we can change it later if it turns out that the representation changes often.